frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/FieldListDrawer/field-panels/PaymentsInputPanel/ProductModal.tsx

```diff
@@ -41,6 +41,12 @@ const MIN_QTY_KEY = `min_qty`
 const MAX_QTY_KEY = `max_qty`
 const DISPLAY_AMOUNT_KEY = 'display_amount'
 const MULTI_QTY_KEY = 'multi_qty'
+
+const parseIntElseNull = (val: string) => {
+  const parsed = parseInt(val, 10)
+  return Number.isNaN(parsed) ? null : parsed
```

Comment:
Just wondering but is NaN truthy? if not then instead of this method we could just `parseInt` then check truthyness right?

Reply:
!!parseInt(`0`, 10) would be falsy, which might not be what the caller expects 🤔.

As of now, we are using the falsy-ness of it, but it might be possible that in other usecase, we might want to do null checks to show some text.

`if (parseIntElseNull(val) === null) { return 'Please enter a number' }`

Reply:
Oh okie! Thanks for the explanation @KenLSM